libtrap/src/ifc_file.c

```diff
@@ -177,49 +176,74 @@ static void file_create_dump(void *priv, uint32_t idx, const char *path)
    free(config_file);
 }
 
-char *create_filename_from_time(void *priv)
+/**
+ * \brief Create a new path and filename from the template created during interface initiation.
+ * \param[in] priv   pointer to module private data
+ * \return 0 on success (TRAP_E_OK), TRAP_E_IO_ERROR if error occurs during directory creation, TRAP_E_BADPARAMS if the specified path and filename exceeds MAX_PATH - 1 bytes.
+ */
+int create_next_filename(void *priv)
 {
-   time_t timer;
-   struct tm* tm_info;
-   time(&timer);
-   tm_info = localtime(&timer);
    file_private_t *config = (file_private_t*) priv;
-   config->starting_time = time(NULL);
-   config->starting_time -= (config->starting_time % (config->file_change_time * 60));
-   config->file_index = 0;
-   char *new_filename = (char*) calloc(config->filename_base_length + 14, sizeof(char));
-   if (!new_filename) {
-      return NULL;
+   char buf[PATH_MAX + 11];
+   config->create_time = time(NULL);
+
+   /* Get actual time a round the time based on user specified parameter */
+   if (config->file_change_time > 0) {
+      config->create_time -= (config->create_time % (config->file_change_time * 60));
    }
 
-   if(strftime(new_filename, config->filename_base_length + 6, config->filename, tm_info) == 0){
-       return NULL;
+   /* Create valid path string based on the template and actaul time */
+   size_t len = strftime(buf, PATH_MAX - 1, config->filename_tmplt, localtime(&config->create_time));
+   if (len == 0) {
+      VERBOSE(CL_ERROR, "FILE IFC[%u]: Path and filename exceeds maximum size: %u.", config->ifc_idx, PATH_MAX - 1);
+      return TRAP_E_BADPARAMS;
    }
-   
-   if(strftime(new_filename + (strlen(new_filename)), 14, ".%Y%m%d%H%M", localtime(&config->starting_time)) == 0){
-       return NULL;
-   };
 
-   return new_filename;
-}
+   /* Recursively create specified directory and subdirectories*/
+   if (_mkdir(buf) != 0) {
+      VERBOSE(CL_ERROR, "FILE IFC[%u]: Unable to create specified directory.", config->ifc_idx);
+      return TRAP_E_IO_ERROR;
+   }
 
-char *create_next_filename(void *priv)
-{
-   file_private_t *config = (file_private_t*) priv;
-   char *new_filename = NULL;
-   char tmp;
-   int offset = (config->file_change_time) ? 13 : 0;
+   /* Create a backup of current path */
+   char backup[PATH_MAX + 11];
```

Comment:
when `PATH_MAX + 11` is repeatedly used, it'd be better to define a constant for it